src/main/java/dao/FileWriterImpl.java

```diff
@@ -0,0 +1,24 @@
+package dao;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class FileWriterImpl implements dao.FileWriter {
+
+    @Override
+    public void writeFile(String fileName, Map<String, Integer> inventory) {
```

Comment:
Consider avoiding hardcoding the file header (`"fruit,quantity"`). You could define it as a constant or pass it as a parameter to allow for more flexible file writing.